tests/single-instance.tftest.hcl

```diff
@@ -0,0 +1,134 @@
+  command = plan
+
+  assert {
+    condition = length(local._root_module_stack_configs) == 1 && local._root_module_stack_configs["root-module-a"] != null
+    error_message = "_root_module_stack_configs is not expected structure: ${jsonencode(local._root_module_stack_configs)}"
+  }
+}
+
+run "test_single_instance_stack_configs_use_default_tf_workspace" {
+  command = plan
+
+  assert {
+    condition = local._root_module_stack_configs["root-module-a"].terraform_workspace == "default"
+    error_message = "terraform_workspace is not set to default: ${jsonencode(local._root_module_stack_configs)}"
+  }
+}
+
+run "test_single_instance_stack_configs_project_root_is_correct" {
+  command = plan
+
+  assert {
+    condition = local._root_module_stack_configs["root-module-a"].project_root == "${var.root_modules_path}/root-module-a"
+    error_message = "project_root is not correct for root-module-a: ${jsonencode(local._root_module_stack_configs)}"
+  }
+}
+
+# Test that the administrative label is not added to the stack when the stack is not set to administrative
+run "test_single_instance_administrative_label_is_not_added_to_stack_when_not_administrative" {
+  command = plan
+
+  assert {
+    condition = !contains(local.labels["root-module-a"], "administrative")
+    error_message = "Administrative label was added to the stack when it should not have been: ${jsonencode(local.labels)}"
+  }
+}
+
+# Test that the depends-on label is added to the stack
+run "test_single_instance_depends_on_label_is_added_to_stack" {
+  command = plan
+
+  assert {
+    condition = contains(local.labels["root-module-a"], "depends-on:spacelift-automation-default")
+    error_message = "Depends-on label was not added to the stack: ${jsonencode(local.labels)}"
+  }
+}
+
+# Test that the folder label is added to the stack and doesn't include a workspace name
+run "test_single_instance_folder_label_is_added_to_stack_and_doesnt_include_workspace_name" {
+  command = plan
+
+  assert {
+    condition = contains(local.labels["root-module-a"], "folder:root-module-a")
+    error_message = "Folder label was not added to the stack: ${jsonencode(local.labels)}"
+  }
+}
+
+# Test that stack.yaml labels are included in the stack labels
+run "test_single_instance_stack_yaml_labels_are_included_in_stack_labels" {
+  command = plan
+
+  assert {
+    condition = contains(local.labels["root-module-a"], "stack_label")
+    error_message = "Stack.yaml labels were not included in the stack labels: ${jsonencode(local.labels)}"
+  }
+}
+
+# Test that the before_init steps are added to the stack
+run "test_single_instance_before_init_steps_are_added_to_stack" {
+  command = plan
+
+  assert {
+    condition = contains(local.before_init["root-module-a"], "echo 'Hello'") && contains(local.before_init["root-module-a"], "echo 'World'")
+    error_message = "Before_init steps were not added to the stack: ${jsonencode(local.before_init)}"
```

Comment:
It is defined in the `before_init` of the config file for that stack, which should be merged here. The test is passing. You probably don't need to comment on test conditions since if they're passing then they're likely correct. And if they're not passing then we'll know about it.